core/corehttp/gateway_handler.go

```diff
@@ -125,6 +125,10 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	i.addUserHeaders(w) // ok, _now_ write user's headers.
 	w.Header().Set("X-IPFS-Path", urlPath)
 
+	//robots-header to prevent search-engines from indexing the gateway because of the duplicate-content problematic when makeing the gateway public
+	w.Header().Set("X-Robots-Tag", "noindex")
```

Comment:
>  I thought the goal for the native-browser implementation is that a browsers understand html-embed like `<img src="ifps://ipfs/[anyhash]" />` instead of binding the protocol to https://ipfs.io ( Like the firefox-ipfs-plugin does )

- Yes, we're moving to a different protocol identifier (both `fs://...` and `ipfs://...` will work)
- The `ipfs.io` gateway **_is an HTTP-to-IPFS proxy_**, and thus a URL like `https://ipfs.io/ipfs/foo` should be taken to mean _exactly_ `/ipfs/foo` which is equivalent to `fs:/ipfs/foo`. we will be declaring this in standards, both for W3C and for IETF, and so on. so don't worry, the `"http://ipfs.io/ipfs"` url prefix will be pretty different than the rest of the network.

> I don't know how familiar you are with SEO but from my knowledge I would refrain from any possibility of creating duplicate content by publishing my content to more than one web-uri.

CDNs, Proxies, Caches are handled by search engines just fine. having the same content in multiple URLs is not _actually_ a problem once things are configured properly (there are lists we need to be included in, etc)

By the way, take a look at http://www.onion.link/ -- it is an HTTP gateway to Tor (like Tor2Web) but which _enables search engine access_ and actually works quite well.

![](http://ipfs.io/ipfs/QmRgxi5LJa8kBe6DC3DkrYwRjvJgxfMEjx18PPM5EknWBS/cap.png)
![](http://ipfs.io/ipfs/QmXbUdtcZR3v6i3zQYURgVTh3QW7anUS6ue6e9tT8gD52H/cap.png)


> I think a better way would be to create a sitemap.xml to display content to SE-crawlers. The ipfs-daemon should also parse the sitemap and add noindex to everything not included in there.

and how do you propose to generate a `sitemap.xml` of _all_ content available in ipfs, including content the gateway does not yet have, but can indeed serve? (hint: you would need a search engine to generate this `sitemap.xml`, and it would function much like google's own index. and you might as well have google do it.)